src/main_thread/init/utils/create_core_playback_observer.ts

```diff
@@ -128,6 +139,10 @@ export default function createCorePlaybackObserver(
       clearSignal: canceller.signal,
       emitCurrentValue: false,
     });
+
+    mediaSource?.addEventListener("streamingChanged", () => {
+      emitCorePlaybackObservation();
+    });
```

Comment:
Yeah why not. We should keep in mind that consequently the playback observations emitted might be a little old by that point. `speed` has the same issue and is prefixed by a `TODO`.

Maybe not in this PR but we could also add a note expliciting this here.

Reply:
> this would have been a memory leak if the event is never removed on cancel, right ?

Yes but it's a little more complex than that, as that closure (the event listener) is then linked to that `MediaSource` Object (basically a `MediaSource` is an `EventTarget` which can be though as the one holding a reference - and thus keeping alive - the listener). If the `MediaSource` is GCed - which can happen once there's no reference left to it - that closure should have no reason to stay in memory.

In any way, it's proper form to explicitly handle listener clean-up at the right time here. That part of the code doesn't know details about MediaSource and event listener management done by the RxPlayer and browser GCs are black boxes anyway that do not always follow the theory.